File: src/java/cleargl/scenegraph/Scene.java
package cleargl.scenegraph;

class Scene extends Node {

  public Scene() {
    super("RootNode");
  }

  public void addNode(Node n, Node parent) {
    if(n.getName().equals("RootNode")) {
      throw new IllegalStateException("Only one RootNode may exist per scenegraph. Please choose a different name.");
    }
  }
}

File: src/java/cleargl/scenegraph/Scene.kt
package cleargl.scenegraph

internal class Scene : Node("RootNode") {

    fun addNode(n: Node, parent: Node) {
        if (n.name == "RootNode") {
            throw IllegalStateException("Only one RootNode may exist per scenegraph. Please choose a different name.")
        }
    }
}